cmd/skaffold/app/cmd/parse_config.go

```diff
@@ -159,27 +166,59 @@ func processEachDependency(d latest.ConfigDependency, fPath string, required boo
 		}
 	}
 	path := d.Path
+
+	if d.GitRepo != nil {
+		cachePath, err := cacheRepo(*d.GitRepo, opts, r)
+		if err != nil {
+			return nil, fmt.Errorf("caching remote dependency %s: %w", d.GitRepo.Repo, err)
+		}
+		path = cachePath
+	}
+
 	if path == "" {
 		// empty path means configs in the same file
 		path = fPath
 	}
 	fi, err := os.Stat(path)
 	if err != nil {
 		if os.IsNotExist(errors.Unwrap(err)) {
-			return nil, fmt.Errorf("could not find skaffold config %s that is referenced as a dependency in config %s", d.Path, fPath)
+			return nil, fmt.Errorf("could not find skaffold config %s that is referenced as a dependency in config %s", path, fPath)
 		}
 		return nil, fmt.Errorf("parsing dependencies for skaffold config %s: %w", fPath, err)
 	}
 	if fi.IsDir() {
 		path = filepath.Join(path, "skaffold.yaml")
 	}
-	depConfigs, err := getConfigs(configOpts{file: path, selection: d.Names, profiles: depProfiles, isRequired: required, isDependency: path != fPath}, opts, appliedProfiles, configNameToFile)
+	depConfigs, err := getConfigs(configOpts{file: path, selection: d.Names, profiles: depProfiles, isRequired: required, isDependency: path != fPath}, opts, r)
 	if err != nil {
 		return nil, err
 	}
 	return depConfigs, nil
 }
 
+// cacheRepo downloads the referenced git repository to skaffold's cache if required and returns the path to the target configuration file in that repository.
+func cacheRepo(g latest.GitInfo, opts config.SkaffoldOptions, r *record) (string, error) {
+	key := fmt.Sprintf("%s:%s", g.Repo, g.Ref)
+	if p, found := r.cachedRepos[key]; found {
+		switch v := p.(type) {
+		case string:
+			return filepath.Join(v, g.Path), nil
+		case error:
+			return "", v
+		default:
+			return "", fmt.Errorf("unable to check download status of repo %s at ref %s", g.Repo, g.Ref)
```

Comment:
This case shouldn't be possible, right?  So it indicates a logical flaw.  Since errors from here don't seem to be recoverable, a `logrus.Fatal()` seems more appropriate.

Reply:
done